tests/queries/0_stateless/01161_all_system_tables.sh

```diff
@@ -23,7 +23,7 @@ function run_selects()
 
     for t in "${tables_arr[@]}"
     do
-        ${CLICKHOUSE_CLIENT} -q "SELECT * FROM $t LIMIT $LIMIT SETTINGS allow_introspection_functions = 1 FORMAT Null" # Suppress style check: database=$CLICKHOUSE_DATABASEs
+        ${CLICKHOUSE_CLIENT} -q "SELECT * FROM $t LIMIT $LIMIT SETTINGS allow_introspection_functions = 1, max_execution_time = 1000 FORMAT Null" # Suppress style check: database=$CLICKHOUSE_DATABASE
```

Comment:
But this will throw an exception in case of failure... Plus 1000 seconds looks too huge

Reply:
Also why does it so slow for metric_log? Yes it is wide, but still, this test has `no-parallel` after all. Maybe some settings randomization?

Reply:
> The test fails with timeout exceeded while fetching system.metric_log. It usually takes 600-700ms, I have set max_execution_time to 1000ms

Hi, @SmitaRKulkarni. @tavplubix and I were also discussing about this flaky test today. From what we looked, it wasn't clear which query that was failing - couldn't find anything that states that it was because of metric_log, at least from the logs that we were looking at. Did you come across any logs that indicated this?

If it's not clear, we can probably add some additional debug information and let it run and catch it the next time. Probably we can use [wait_for_queries_to_finish](https://github.com/ClickHouse/ClickHouse/blob/77e65df0fd64977ba0f26679def669dac2ac100b/tests/queries/shell_config.sh#L159 ) I think?

Reply:
>Probably we can use [wait_for_queries_to_finish](https://github.com/ClickHouse/ClickHouse/blob/77e65df0fd64977ba0f26679def669dac2ac100b/tests/queries/shell_config.sh#L159) I think?

Hm, how this will help if query just slow?

Reply:
I have one idea - https://github.com/ClickHouse/ClickHouse/pull/79013

Reply:
And I also have a proposal on how to improve introspection of .sh tests - https://github.com/ClickHouse/ClickHouse/issues/78627, and since this pops up again, I've implemented it - https://github.com/ClickHouse/ClickHouse/pull/79015